Add config for Atreus based on Adafruit Feather BLE

ClashTheBunny opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
This adds both a rules.mk and a keyboard layout for an Atreus converted to Bluetooth using an Adafruit Feather BLE.  The README in the `/feather/` directory includes directions and pictures of how to solder on the new microcontroller.  The keymap is pretty much the default keyboard layout with the bluetooth switches and a mouse added to layer three.

This is based on the normal bluetooth support and just works, so it isn't much to maintain.  I use this keyboard as my commuting keyboard and it works great.
<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

zvecr commented on keyboards/atreus/feather/rules.mk:
```diff
@@ -0,0 +1,35 @@
+# MCU name
+MCU = atmega32u4
+F_CPU = 8000000
+
+BOOTMAGIC_ENABLE = no         # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes         # Mouse keys(+4700)
+# EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no           # Console for debug(+400)
+COMMAND_ENABLE = no           # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+# SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+# NKRO_ENABLE = yes           # USB Nkey Rollover
+# BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+# RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+# MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = yes          # Unicode
+
+BLUETOOTH_ENABLE = yes
+BLUETOOTH = AdafruitBLE
+OUTPUT_AUTO_ENABLE = yes
+
+# AUDIO_ENABLE = no           # Audio output on port C6
+# FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+# HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
```
```suggestion

```

zvecr commented on keyboards/atreus/feather/rules.mk:
```diff
@@ -0,0 +1,35 @@
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
```
```suggestion
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BLUETOOTH = AdafruitBLE
```

ClashTheBunny replied:
Is this trying to get this to be a "base" so that people can add the normal options if they wish?  It seems like we would at least want `OUTPUT_AUTO_ENABLE = yes`.  Maybe `MOUSEKEY_ENABLE = yes         # Mouse keys(+4700)` would be required to use the clash keyboard layout?  Are the rest defaults that don't actually affect output code?  I know that mousekey and output_auto are usually false as well.

zvecr replied:
OUTPUT_AUTO_ENABLE isnt a valid `rules.mk` option, it would have to go in a `config.h` for it to have any effect. As for mousekeys, it is enabled at the parent level <https://github.com/qmk/qmk_firmware/blob/master/keyboards/atreus/rules.mk#L5>.

This new keyboard revision should only set the minimal options required for it to function, as per the rest of the other revisions.

fauxpark commented on keyboards/atreus/feather/rules.mk:
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = atmega32u4
+F_CPU = 8000000
+# Bootloader selection
```
```suggestion

# Processor frequency
F_CPU = 8000000

# Bootloader selection
```

fauxpark commented on keyboards/atreus/keymaps/clash/keymap.c:
```diff
@@ -0,0 +1,44 @@
+// this is the style you want to emulate.
+// This is the canonical layout file for the Quantum project. If you want to add another keyboard,
+
```
```suggestion
```

fauxpark commented on keyboards/atreus/feather/rules.mk:
```diff
@@ -0,0 +1,21 @@
+#   change yes to no to disable
+#
+BLUETOOTH = AdafruitBLE
+CONSOLE_ENABLE = no           # Console for debug(-400)
```
```suggestion
CONSOLE_ENABLE = no
```

ClashTheBunny commented:
With the suggested changes, I get that the firmware is too large:
```
Checking file size of atreus_feather_clash.hex
 * The firmware is too large! 30690/28672 (2018 bytes over)
 [ERRORS]

make[1]: *** [tmk_core/rules.mk:411: check-size] Error 1
make: *** [Makefile:582: atreus/feather:clash:avrdude] Error 1
Make finished with errors
```
I removed CONSOLE and it's small enough.

I also added `OUTPUT_AUTO_ENABLE` to the `config.h`, as mentioned by @zvecr.

drashna commented on keyboards/atreus/feather/readme.md:
````diff
@@ -0,0 +1,60 @@
+To build it and flash it, do:
+
+```
+make atreus/feather:clash:avrdude
````
```suggestion
make atreus/feather:clash:flash
```

ClashTheBunny commented:
So, can we merge this?

fauxpark commented:
Thanks!
